test/unit/lib/plugins/aws/package/lib/mergeIamTemplates.test.js

```diff
@@ -245,6 +263,108 @@ describe('lib/plugins/aws/package/lib/mergeIamTemplates.test.js', () => {
+                    {
+                      Effect: 'Allow',
+                      Resource: '*',
+                      NotAction: 'iam:DeleteUser',
+                    },
+                  ],
+                  managedPolicies: [
+                    'arn:aws:iam::123456789012:user/*',
+                    'arn:aws:s3:::my_corporate_bucket/Development/*',
+                    'arn:aws:iam::123456789012:u*',
+                  ],
+                  permissionBoundary: ['arn:aws:iam::123456789012:policy/XCompanyBoundaries'],
+                },
+              },
+              vpc: {
+                securityGroupIds: ['xxx'],
+                subnetIds: ['xxx'],
+              },
+              logRetentionInDays: 5,
+            },
+          },
+        });
+
+        cfResources = cfTemplate.Resources;
+        naming = awsNaming;
+        service = fixtureData.serviceConfig.service;
+      });
+
+      it('should support `provider.iam.role.statements`', async () => {
+        const IamRoleLambdaExecution = naming.getRoleLogicalId();
+        const iamResource = cfResources[IamRoleLambdaExecution];
+        const { Statement } = iamResource.Properties.Policies[0].PolicyDocument;
+
+        expect(Statement).to.deep.includes({
+          Effect: 'Allow',
+          Resource: '*',
+          NotAction: ['iam:DeleteUser'],
+        });
+      });
+
+      it('should support `provider.iam.role.managedPolicies`', () => {
+        const IamRoleLambdaExecution = naming.getRoleLogicalId();
+        const {
+          Properties: { ManagedPolicyArns },
+        } = cfResources[IamRoleLambdaExecution];
+
+        expect(ManagedPolicyArns).to.deep.includes('arn:aws:iam::123456789012:user/*');
+        expect(ManagedPolicyArns).to.deep.includes(
+          'arn:aws:s3:::my_corporate_bucket/Development/*'
+        );
+        expect(ManagedPolicyArns).to.deep.includes('arn:aws:iam::123456789012:u*');
+      });
+
+      it('should support `provider.iam.role.permissionsBoundary`', () => {
+        const IamRoleLambdaExecution = naming.getRoleLogicalId();
+        const {
+          Properties: { PermissionsBoundary },
+        } = cfResources[IamRoleLambdaExecution];
+        expect(PermissionsBoundary).to.be.equal(
+          'arn:aws:iam::123456789012:policy/XCompanyBoundaries'
+        );
+      });
+
+      it('should ensure needed IAM configuration when `provider.vpc` is configured', () => {
+        const IamRoleLambdaExecution = naming.getRoleLogicalId();
+        const iamResource = cfResources[IamRoleLambdaExecution];
+
+        expect(iamResource.Properties.ManagedPolicyArns).to.deep.includes({
+          'Fn::Join': [
+            '',
+            [
+              'arn:',
+              { Ref: 'AWS::Partition' },
+              ':iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole',
+            ],
+          ],
+        });
+      });
+
+      it('should support `provider.logRetentionInDays`', () => {
```

Comment:
Is there a reason this was included as a part of this PR or it's a test unrelated to main changes?

Reply:
Ops.